src/Commands/Visualize.php

```diff
@@ -0,0 +1,136 @@
+
+    protected $config;
+
+    /**
+     * Create a new command instance.
+     *
+     * @param array $config
+     */
+    public function __construct(array $config)
+    {
+        parent::__construct();
+
+        $this->config = $config;
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return mixed
+     */
+    public function handle()
+    {
+        if (empty($this->config)) {
+            $this->error('There are no state machines configured.');
+
+            return 1;
+        }
+
+        if (! $this->argument('graph')) {
+            $this->askForGraph();
+        }
+
+        $graph = $this->argument('graph');
+
+        if (! array_key_exists($graph, $this->config)) {
+            $this->error('The provided state machine graph is not configured.');
+
+            return 1;
+        }
+
+        $config = $this->config[$graph];
+
+        $this->stateMachineInDotFormat($config);
+
+        return 0;
+    }
+
+    protected function stateMachineInDotFormat(array $config)
+    {
+        // Output image mime types.
+        $mimeTypes = [
+            'png' => 'image/png',
+            'jpg' => 'image/jpeg',
+            'gif' => 'image/gif',
+            'svg' => 'image/svg+xml',
+        ];
+
+        $format = $this->option('format');
+
+        if (empty($mimeTypes[$format])) {
+            throw new \Exception(sprintf("Format '%s' is not supported", $format));
+        }
+
+        $dotPath = $this->option('dot-path');
+
+        // Temporary files.
+        $dotFile = tempnam(sys_get_temp_dir(), 'smv');
+        $outputImage = $this->option('output');
+
+        // Display settings
+        $layout = $this->option('direction');
+        $layout = $layout === 'TB' ? 'TB' : 'LR';
+
+        $nodeShape = $this->option('shape');
+
+        // Build dot file content.
+        $result = [];
+        $result[] = 'digraph finite_state_machine {';
+        $result[] = "rankdir=$layout;";
+        $result[] = 'node [shape = point]; _start_'; // Input node
+
+        // Use first value from 'states' as start.
+        $start = $config['states'][0]['name'];
+        $result[] = "node [shape = $nodeShape];"; // Default nodes
+        $result[] = '_start_ -> '.$start.';'; // Input node -> starting node.
+
+        foreach ($config['transitions'] as $name => $transition) {
+            foreach ($transition['from'] as $from) {
+                // TODO: Add customization.
+                $result[] = $from.' -> '.$transition['to'].'[ label = "'.$name.'" ];';
+            }
+        }
+
+        $result[] = '}';
+
+        $result = implode(PHP_EOL, $result);
+
+        // Save dot file for input.
+        file_put_contents($dotFile, $result);
+
+        // Dot command.
+        $cmd = sprintf(
+            "%s -T%s -o %s %s",
+            $dotPath,
+            $format,
+            $outputImage, // Output file
+            $dotFile // Input file
+        );
+
+
+        $process = new Process($cmd);
```

Comment:
One thing we should do is require `https://github.com/symfony/Process` since we need that to execute `dot`.

Also it would be better to use the Process like in the documentation: https://symfony.com/doc/current/components/process.html#usage
(Ysing an array instead of the `$cmd` string)

Reply:
You're right, done